pnpm/test/hooks.ts

```diff
@@ -223,3 +224,75 @@ test('custom fetcher can call default fetcher', async () => {
 
   expect(args.opts).toBeDefined()
 })
+
+test('adding or changing pnpmfile should change pnpmfileChecksum and module structure', async () => {
+  const project = prepare({
+    dependencies: {
+      '@pnpm.e2e/pkg-with-good-optional': '1.0.0',
+    },
+  })
+
+  await execPnpm(['install'])
+
+  const lockfile0 = project.readLockfile()
+  expect(lockfile0.pnpmfileChecksum).toBeUndefined()
+  expect(lockfile0.packages).toHaveProperty(['/@pnpm.e2e/pkg-with-good-optional@1.0.0'])
+  expect(lockfile0.packages).toHaveProperty(['/is-positive@1.0.0'])
+
+  const pnpmfile1 = `
+    function readPackage (pkg) {
+      if (pkg.optionalDependencies) {
+        pkg.optionalDependencies = {} // 'delete pkg.optionalDependencies' doesn't work here
+      }
+      return pkg
+    }
+
+    module.exports.hooks = { readPackage }
+  `
+  fs.writeFileSync('.pnpmfile.cjs', pnpmfile1)
+  await execPnpm(['install'])
+
+  const lockfile1 = project.readLockfile()
+  expect(lockfile1.pnpmfileChecksum).toBe(createBase32Hash(pnpmfile1))
+  expect(lockfile1.packages).toHaveProperty(['/@pnpm.e2e/pkg-with-good-optional@1.0.0'])
+  expect(lockfile1.packages).not.toHaveProperty(['/is-positive@1.0.0']) // this should be removed due to being optional dependency
```

Comment:
Did you say that `.toHaveProperty` doesn't work correctly when the key has dots?

Reply:
Not when it's an item in an array.